catalogs/usfimr-s1/build_catalog.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
+
+from datetime import datetime
+import json
+import argparse
+from itertools import groupby
+
+from pystac import (
+    Asset,
+    CatalogType,
+    Catalog,
+    Collection,
+    Extent,
+    Item,
+    SpatialExtent,
+    TemporalExtent,
+)
+import boto3
+import urllib3
+import rasterio as rio
+from shapely.geometry import Polygon, mapping
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--imagery-root-s3", required=True)
+    args = parser.parse_args()
+
+    parsed_s3_path = urllib3.util.parse_url(args.imagery_root_s3)
+    bucket = parsed_s3_path.netloc
+    prefix = parsed_s3_path.path[1:]
+
+    s3 = boto3.resource("s3")
+    bucket = s3.Bucket(bucket)
+    filtered_objects = bucket.objects.filter(Prefix=prefix)
+
+    catalog_description = (
+        "Sentinel-1 imagery corresponding to flood events catalogued within GLOFIMR"
+    )
+    catalog_title = "GLOFIMR SAR Imagery"
+
+    catalog = Catalog("glofimr-sar", catalog_description, title=catalog_title)
+
+    # We know the IDs used here, so we'll use them for keys
```

Comment:
Man it sucks that GLOFIMR doesn't provide a consistent, useful id. 

Perhaps remove the TODO now that this is being merged and replace with an extra comment stating that this numeric id corresponds to the incrementing id in `ftp://guest:guest@sdml-server.ua.edu/USFIMR/USFIMR_all.zip` so future us can at least trace back to the actual events we're referring to here.